core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -47,6 +48,25 @@ protected BaseMetadataTable(TableOperations ops, Table table, String name) {
     this.name = name;
   }
 
+
+  /**
+   * This method transforms the table's partition spec to a spec that is used to rewrite the user-provided filter
+   * expression against the partitions table.
+   * <p>
+   * The resulting partition spec maps partition.X fields to partition X using an identity partition transform. When
+   * this spec is used to project an expression for the partitions table, the projection will remove predicates for
+   * non-partition fields (not in the spec) and will remove the "partition." prefix from fields.
+   *
+   * @param partitionTableSchema schema of the partition table
```

Comment:
Looks like this is still assuming the partition table. May want to update it to `tableSchema`.

While we're thinking about this, it may also make sense to allow passing a different prefix. The prefix for the entries table, for example, would be `data_file.partition.`

Reply:
Good catch, rewrote the comment and added an argument.